core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -589,4 +616,56 @@ object StorageTool extends Logging {
       (nameAndLevel._1, nameAndLevel._2)
     }.toMap
   }
-}
+
+  def writeCheckpointFile(stream: PrintStream, logDir: String, directoryId: Uuid,
+                          advertisedListenerEndpoints: scala.collection.Seq[kafka.cluster.EndPoint]): Unit = {
+    val snapshotCheckpointDir = logDir + clusterMetadataDir
+
+    // Ensure the directory exists
+    val snapshotDir = Paths.get(snapshotCheckpointDir)
+    if (!Files.exists(snapshotDir)) {
+      Files.createDirectories(snapshotDir)
+    }
+
+    // Create the full path for the checkpoint file
+    val checkpointFilePath = snapshotDir.resolve(snapshotDir)
+
+    // Create the raw snapshot writer
+    val rawSnapshotWriter = FileRawSnapshotWriter.create(checkpointFilePath, new OffsetAndEpoch(0, 0))
```

Comment:
Instead of hard-coding `OffsetAndEpoch(0, 0)` let's add a constant to `o.a.k.s.Snapshots` named `BOOTSTRAP_SNAPSHOT_ID`.